fff

```diff
@@ -368,6 +369,50 @@ trash() {
     fi
 }
 
+bulk_rename() {
+    # Bulk rename files using '$EDITOR'.
+    rename_file="${XDG_CACHE_HOME:=${HOME}/.cache}/fff/bulk_rename"
+    marked_files=("${@:1:$#-1}")
+
+    # Save marked files to a file and open them for editing.
+    printf '%s\n' "${marked_files[@]}" > "$rename_file"
+    "${VISUAL:-${EDITOR:-vi}}" "$rename_file"
+
+    # Read the renamed files to an array.
+    IFS=$'\n' read -d "" -ra changed_files < "$rename_file"
+
+    # If the user deleted a line, stop here.
+    ((${#marked_files[@]} != ${#changed_files[@]})) && {
+        rm "$rename_file"
+        cmd_line "warn: Line mismatch in rename file. Doing nothing."
+        return
+    }
+
+    printf '%s\n%s\n' \
+        "# This file will be executed when the editor is closed." \
+        "# Clear the file to abort." > "$rename_file"
+
+    # Construct the rename commands.
+    for ((i=0;i<${#marked_files[@]};i++)); {
+        [[ ${marked_files[i]} != "${changed_files[i]}" ]] && {
+            printf 'mv -- "%s" "%s"\n' "${marked_files[i]}" "${changed_files[i]}"
```

Comment:
I guess we still need -i flag, so user doesn't overwrite already existing file by accident

Reply:
Check the latest commit. I swapped to `[[ ! -f ${changed_files[i]} ]]` (if the new filename doesn't already exist, add the command to the list).